CenterForOpenScience/osf.io
# Feature/Deactivation_requested Field User Settings v2 [PLAT-1312]

## Purpose
The `/v2/users/me/settings/` endpoint needs a field `deactivation_requested` (boolean) that indicates if a user has requested deactivation on their account. Also, we need to be able to undo that request. PATCH the user settings endpoint and change deactivation_requested that way. We also need a `Link` in the settings object for `export` that points to `/v2/users/<user-id>/settings/export/`

## Changes

- Adds a `deactivation_requested` attribute to the UserSerializer for marking User.requested_deactivation true/false.  You can set either way.
- Setting as True sends an email to osf support w/ the request_deactivation info
- Setting as False, just marks as False.
- Endpoint is throttled
- Removes similar deactivate endpoint introduced in https://github.com/CenterForOpenScience/osf.io/pull/8527 
- Exposes `export` link in UserSettings

![screen shot 2019-01-23 at 8 12 30 am](https://user-images.githubusercontent.com/9755598/51615332-2b0f0d80-1eed-11e9-88fe-d1bf34f4e038.png)



## QA Notes

API only - can double check that you can set deactivation_requested to True/False in a PATCH to users/user_id/settings/. And that the requests are throttled.

## Documentation

Probably not documenting this one.

## Side Effects



## Ticket

https://openscience.atlassian.net/browse/PLAT-1312
